From 2e16b32d484c0907cd08a50383745b5dc341e51e Mon Sep 17 00:00:00 2001
From: Sanyog Kale <skale@habana.ai>
Date: Thu, 26 Nov 2020 07:24:38 +0200
Subject: [PATCH] [SW-22029] recovery: disable pcie error responses

By default, PCIe error responses for LBW and HBW are enabled in
recovery mode which may lead to PCIe link down in case error are seen
on LBW and HBW. In order to avoid PCIe link down, this patch add changes
to disable error responses for LBW and HBW in recovery mode.

Error responses are disabled by setting value 0x1 in
mmPCIE_WRAP_PCIE_RSP_OK register.

Change-Id: I26072268d5ab6ebd9a664af648525dc9257cc8e8
---

diff --git a/common/pcie.c b/common/pcie.c
index 332fcd4..5e6fdf3 100644
--- a/common/pcie.c
+++ b/common/pcie.c
@@ -208,6 +208,9 @@
 		regmap_writel(spare0, mmPCIE_WRAP_SPECIAL_GLBL_SPARE_0);
 	}
 
+	/* Disable PCIe error responses (both HBW and LBW) */
+	regmap_writel(0x1, mmPCIE_WRAP_PCIE_RSP_OK);
+
 	security_debug_if_secure();
 	security_debug_if_shared();
 
